File: source_funcs/go_transcoder/SUBSEQUENCES_SIZE_THREE_ARRAY_WHOSE_SUM_DIVISIBLE_M/SUBSEQUENCES_SIZE_THREE_ARRAY_WHOSE_SUM_DIVISIBLE_M_emit.go
package main

//go:export f_gold
func f_gold(A []int, N int, M int) int {
	var (
		sum int = 0
		ans int = 0
	)
	for i := int(0); i < N; i++ {
		for j := int(i + 1); j < N; j++ {
			for k := int(j + 1); k < N; k++ {
				sum = A[i] + A[j] + A[k]
				if sum%M == 0 {
					ans++
				}
			}
		}
	}
	return ans
}

File: rust_funcs/go_transcoder/SUBSEQUENCES_SIZE_THREE_ARRAY_WHOSE_SUM_DIVISIBLE_M/SUBSEQUENCES_SIZE_THREE_ARRAY_WHOSE_SUM_DIVISIBLE_M_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


#[no_mangle]
fn f_gold(A: [i32;2], N: i32, M: i32) -> i32 {
    let mut sum: i32;
    let mut ans: i32 = 0;
    for i in 0..N {
        for j in (i + 1)..N {
            for k in (j + 1)..N {
                sum = A[i as usize] + A[j as usize] + A[k as usize];
                if sum % M == 0 {
                    ans += 1;
                }
            }
        }
    }
    ans
}
